### PR TITLE
5 make favourite controller

### LINKED ISSUE #5: Make favourite controller
Make favourite controller in which:
1. User can add kebab to favourite. User can't add two same kebabs to favourite.
2. User can remove kebab from favourite
3. User can get all HIS favourite kebabs.

### PR DESCRIPTION
**Pull request by W-Worach**
#5  Added favourite controller with functions
 - add to favourites
 - remove from favourites
 - get user favourites

### CODE REVIEW

File: KebabBackend/app/Http/Controllers/Api/FavouriteController.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public function __construct(FavouriteService $favouriteService)
+    {
+        $this->favouriteService = $favouriteService;
+    }
+
+    /**
+     * Add a kebab to the authenticated user's favourites.
+     *
+     * @OA\Post(
+     *     path="/kebabs/{kebab}/favourite",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/FavouriteController.php]:
```suggestion
     *     path="/api/kebabs/{kebab}/favourite",
```

File: KebabBackend/app/Http/Controllers/Api/FavouriteController.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        return response()->json(['message' => 'Kebab added to favourites'], 201);
+    }
+
+    /**
+     * Remove a kebab from the authenticated user's favourites.
+     *
+     * @OA\Delete(
+     *     path="/kebabs/{kebab}/favourite",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/FavouriteController.php]:
```suggestion
     *     path="/api/kebabs/{kebab}/favourite",
```

File: KebabBackend/app/Http/Controllers/Api/FavouriteController.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        return response()->json(null, 204);
+    }
+
+    /**
+     * Get all favourite kebabs of the authenticated user.
+     *
+     * @OA\Get(
+     *     path="/user/favourites",
```
Comment (Wiktor-Wojciechowski) [inline KebabBackend/app/Http/Controllers/Api/FavouriteController.php]:
```suggestion
     *     path="/api/user/favourites",
```

**Review by Wiktor-Wojciechowski — CHANGES_REQUESTED** [review]
I suggest changing these paths in the docs to fit with the rest and keep them consistent.
Remember to pull from main before merging too as this branch does not have the updated paths in other places

**Review by MatWisie — APPROVED** [review]
New, favourite routes, are not visible in swagger documentation. Include it in documentation and its good to go
obraz

**Review by Wiktor-Wojciechowski — APPROVED** [review]

**Pull request merged**